### ISSUE TITLE
Rule request: report unnecessary lint suppression line comments

### ISSUE THREAD

**Issue by niedzielski** [eslint/eslint#7872]
**Please describe what the rule should do:**

This rule should report unnecessary lint suppression line comments which can accrue over time and need to be maintained.

**What category of rule is this? (place an "X" next to just one item)**

[x] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

👍 Good, ESLint would flag this line as too long:
```js
/* eslint max-len: "error"]*/

// eslint-disable-next-line max-len
const foo = 'abcdefghijklmnopqrstuvwxyz0123456789abcdefghijklmnopqrstuvwxyz0123456789abcdefghijklmnopqrstuvwxyz0123456789abcdefghijklmnopqrstuvwxyz0123456789'
```

👎 Bad, suppression is outdated and no longer necessary:
```js
/* eslint max-len: "error"]*/

// eslint-disable-next-line max-len
const foo = 'abc'
```

**Why should this rule be included in ESLint (instead of a plugin)?** This rule would be universally beneficial to all other suppressable rules

**Comment by niedzielski** [comment]
@gyandeeps, thanks :] I think this can be closed as a duplicate
